### PR TITLE
fix TestSignal test

### PR DESCRIPTION
**Pull request by mkumatag**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Fixes: #14846

### CODE REVIEW

**Comment by Rajalakshmi-Girish** [comment]
\lgtm

File: pkg/expect/expect_test.go [hunk 1]
```diff
@@ -189,7 +190,7 @@ func TestSignal(t *testing.T) {
 	go func() {
 		defer close(donec)
 		err = ep.Close()
-		assert.ErrorContains(t, err, "unexpected exit code [-1] after running [/usr/bin/sleep 100]")
+		assert.Regexp(t, regexp.MustCompile(`unexpected exit code \[-1\] after running \[.*sleep 100\]`), err)
```
Comment (serathius) [inline pkg/expect/expect_test.go:193]:
```suggestion
		assert.ErrorContains(t, err, "unexpected exit code [-1]")
```
@tjungblu to confirm

Comment (mkumatag) [reply pkg/expect/expect_test.go:193]: I don't mind accepting the suggestion and ignore the rest of the message :) lets see what @tjungblu says.!

Comment (ptabor) [reply pkg/expect/expect_test.go:193]: We don't care about the path... but I think that it's convenient that expect writes which command failed,
so I'm on the side of keeping check whether `sleep 100` is mentioned. 

The original change or: 
```
assert.ErrorContains(t, err, "unexpected exit code [-1]")
assert.ErrorContains(t, err, "sleep 100")
```
both works for me.

Comment (mkumatag) [reply pkg/expect/expect_test.go:193]: I feel multiple asserts are better and easy to read as well, lemme adopt them instead of regex.

**Review by ptabor — APPROVED** [review]

**Comment by ptabor** [comment]
I see @serathius thumb-up on my comment, so I think its ok to merge. Thank you @mkumatag .

**Pull request merged**